### ISSUE TITLE
Biome Specific Spawning in 1.18.2

### ISSUE THREAD

**Issue by DeliciousTea** [McJty/InControl#369]
I've seen some other threads on this and haven't been able to figure out how to make this work myself. Here's the breakdown:

I'm adding a mob that doesn't have configurable spawns to the Spawner.json in InControl like so:

{
    "mob": ["enemyexpansion:caster"],
    "persecond": 1,
    "attempts": 10,
	"weights": 80,
    "amount": {
      "minimum": 1,
      "maximum": 2,
      "groupdistance": 3
    },
    "conditions": {
      "dimension": "bro:saturated",
      "inwater": false,
      "inair": false,
      "mindist": 40,
      "maxdist": 128,
      "minheight": 60,
      "maxheight": 160,
      "maxhostile": 60
    }
  },

This part works fine, the mob spawns in the custom dimension, they spawn in every custom biome. I need to force them to spawn in ONLY the biomes I want to specify and I'm trying to do that using the Spawn.json file like this: 

 {
	"mob": "enemyexpansion:caster",
    "biome": "bro:saturated/the_enchanted",
    "result": "default"
  },
  {
	"mob": "enemyexpansion:caster",
    "biome": "bro:saturated/stranger_flats",
    "result": "deny"
  }

One rule allows them in a certain biome, the other denies them in a different biome. The problem is that this doesn't work at all, the mob still spawns in every biome in the custom dimension. I've tried a bunch of different configurations, I've checked the biome names, nothing seems to work. Any help would be appreciated. Thank you!

**Comment by McJty** [comment]
Remove "biome": "bro:saturated/stranger_flats",
 in the second rule. You want to deny in all biomes except the first

**Comment by DeliciousTea** [comment]
Thanks for the response! I had tried that before and was still seeing mobs spawning outside of the specified biome. I just tried it again and they're still spawning in every biome in this custom dimension. The Spawn.json file now reads:

  {
	"mob": "enemyexpansion:caster",
    "biome": "bro:saturated/the_enchanted",
    "result": "default"
  },
  {
	"mob": "enemyexpansion:caster",
    "result": "deny"
  }

Haven't changed the Spawner.json file.

I've tried a bunch of different configurations, removing the dimension condition from Spawner.json results in no spawns in the dimension at all. Using "onjoin": true  in the deny result rule stops all spawns everywhere no matter what. I'm not sure if there's a way to solve this but wanted to bring it to your attention anyway.
